### DISCUSSION TITLE
Control vector in algebraic equations

### DISCUSSION THREAD

**Discussion by rolfjl** [casadi/casadi#3780] [Q&A] [answered]
Dear all, 

I have a DAE system where the algebraic equations also depend on the control vector _u._ According to the documentation (section 4.4 in the online manual), this is supported in CasADi. However, when I run the simple test case given below ($\dot{x} = u, z - u = 0$) I get an error message. If I remove _u_ from the algebraic expression, the code is running. Any suggestions on how to implement this is highly appreciated. 

Rolf

Test code:
```
from casadi import * 
import numpy as np

T = [i*0.1 for i in range(10)]

x = MX.sym('x')
u = MX.sym('u')
z = MX.sym('z')

dae = {'x': x, 'z': z, 'u': u, 'ode': u, 'alg': z-u}
F = integrator('F', 'idas', dae, 0, T)

u_in = np.ones((1,len(T)))
r = F(x0=0, z0=1, u=u_in) 
```

Error message: 

> At t = 0 and h = 3.8147e-10, the error test failed repeatedly or with |h| = hmin.
Traceback (most recent call last):
  File "c:\Users\rolo\Python-Ensemble-Toolbox\RAS\control_test.py", line 14, in <module>
    r = F(x0=0, z0=1, u=u_in)
        ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\rolo\Python-Ensemble-Toolbox\venv\Lib\site-packages\casadi\casadi.py", line 23382, in __call__
    return self.call(kwargs)
           ^^^^^^^^^^^^^^^^^
  File "C:\Users\rolo\Python-Ensemble-Toolbox\venv\Lib\site-packages\casadi\casadi.py", line 20031, in call
    return _casadi.Function_call(self, *args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
RuntimeError: Error in Function::call for 'F' [IdasInterface] at .../casadi/core/function.cpp:1432:
Error in Function::call for 'F' [IdasInterface] at .../casadi/core/function.cpp:361:
.../casadi/interfaces/sundials/idas_interface.cpp:596: IDASolve returned "IDA_ERR_FAIL". Consult IDAS documentation.

**Comment by tmmsartor** [comment, marked as answer]
You may try to use `suppress_algebraic` if works for you, seems to work for the example you provided.
Docs says "Suppress algebraic variables in the error testing",
more information [here](https://sundials.readthedocs.io/en/latest/ida/Usage/index.html#c.IDASetSuppressAlg)

You can set the option like in the following code:

```
from casadi import * 
import numpy as np

N = 10
T = 1.0
tgrid = np.linspace(0,T,N+1)

x = MX.sym('x')
u = MX.sym('u')
z = MX.sym('z')

opts = {}
opts["suppress_algebraic"] = True
dae = {'x': x, 'z': z, 'u': u, 'ode': u, 'alg':z-u}
F = integrator('F', 'idas', dae, tgrid[0], tgrid[1:], opts)

u_in = np.ones(N)
r = F(x0=0.0, z0=1.0, u=u_in)
print(r["xf"])
print(r["zf"])
```

**Reply by rolfjl** [reply]
This solves the problem. 

Thanks a lot!

Rolf
